oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandlerTest.java

```diff
@@ -1693,6 +1694,136 @@ void shouldErrorIfIdTokenHasIncorrectClient() throws JOSEException {
+                if (browserSessionIdFromCookie != null) {
+                    event.setHeaders(Map.of("browser-session-id", browserSessionIdFromCookie));
+                }
+
+                return makeHandlerRequest(event);
+            }
+
+            String browserSessionIdCookieFromResponse(APIGatewayProxyResponseEvent response) {
+                return response.getMultiValueHeaders().get(ResponseHeaders.SET_COOKIE).get(2);
+            }
+
+            @Test
+            void noExistingSessionAndNoBrowserSessionIdInHeaders() {
+                ArgumentCaptor<Session> sessionCaptor = ArgumentCaptor.forClass(Session.class);
+                APIGatewayProxyResponseEvent response = setupWithExistingSession(null, null);
+
+                verify(sessionService).createSession();
+                verify(sessionService).save(sessionCaptor.capture());
+                assertEquals("a-new-session-id", sessionCaptor.getValue().getSessionId());
+                assertEquals(
+                        "a-new-browser-session-id", sessionCaptor.getValue().getBrowserSessionId());
+                assertEquals(
+                        "browser-session-id=a-new-browser-session-id; Domain=auth.ida.digital.cabinet-office.gov.uk; Secure; HttpOnly;",
+                        browserSessionIdCookieFromResponse(response));
+            }
+
+            @Test
+            void noExistingSessionAndBrowserSessionIdInHeaders() {
+                ArgumentCaptor<Session> sessionCaptor = ArgumentCaptor.forClass(Session.class);
+                APIGatewayProxyResponseEvent response =
+                        setupWithExistingSession(null, "id-from-cookie");
+
+                verify(sessionService).createSession();
+                verify(sessionService).save(sessionCaptor.capture());
+                assertEquals("a-new-session-id", sessionCaptor.getValue().getSessionId());
+                assertEquals(
+                        "a-new-browser-session-id", sessionCaptor.getValue().getBrowserSessionId());
+                assertEquals(
+                        "browser-session-id=a-new-browser-session-id; Domain=auth.ida.digital.cabinet-office.gov.uk; Secure; HttpOnly;",
+                        browserSessionIdCookieFromResponse(response));
+            }
+
+            @Test
+            void existingSessionAndNoBrowserSessionIdInHeaders() {
+                ArgumentCaptor<Session> sessionCaptor = ArgumentCaptor.forClass(Session.class);
+                APIGatewayProxyResponseEvent response =
+                        setupWithExistingSession(
+                                new Session("existing-session-id").withBrowserSessionId("same-id"),
+                                null);
+
+                verify(sessionService, never()).createSession();
+                verify(sessionService).save(sessionCaptor.capture());
+                assertEquals("existing-session-id", sessionCaptor.getValue().getSessionId());
```

Comment:
This feels like an odd thing to test, because we also cycle the session ID, right? This test only passes because we've mocked `sessionService.updateSessionId` to be a noop - it doesn't actually describe the desired system behaviour

Reply:
Yea I thought about this too. If a user has a session and they have the correct ids in their request to use it, we always cycle the `sessionId` to something new (in this test, we don't, so it's remaining as "existing-session-id"); if not, we generate a new one ("a-new-session-id"). So whether the cycler is mocked or does nothing to the existing session, we still have a 1:1 map of outcomes (I think?), and the session cycler isn't what's being tested. But that's not clear here. 

I think it makes it readable for when we trigger a new session to be created, but that is already tested by the session's browser session id. What do you reckon?

Reply:
This makes sense. Happy to keep it, just want to make sure it's been considered properly